FinSearchUnified/Tests/BusinessLogic/Models/FindologicArticleModelTest.php

```diff
@@ -1044,4 +1045,134 @@ public function testTranslatedBooleanProperties($highlight, $locale, $expectedVa
+            ],
+            'mainDetail' => [
+                'number' => 'FINDOLOGIC1',
+                'active' => true,
+                'prices' => [
+                    [
+                        'customerGroupKey' => 'EK',
+                        'price' => 130,
+                    ],
+                ]
+            ],
+            'configuratorSet' => [
+                'groups' => []
+            ],
+            'variants' => [
+                [
+                    'isMain' => false,
+                    'number' => 'FINDOLOGIC1.1',
+                    'active' => true,
+                    'inStock' => 0,
+                    'lastStock' => false,
+                    'prices' => [
+                        [
+                            'customerGroupKey' => 'EK',
+                            'price' => 110,
+                        ],
+                    ],
+                    'configuratorOptions' => []
+                ],
+                [
+                    'isMain' => false,
+                    'number' => 'FINDOLOGIC1.2',
+                    'active' => true,
+                    'inStock' => 5,
+                    'lastStock' => true,
+                    'prices' => [
+                        [
+                            'customerGroupKey' => 'EK',
+                            'price' => 120,
+                        ],
+                    ],
+                    'configuratorOptions' => []
+                ],
+                [
+                    'isMain' => false,
+                    'number' => 'FINDOLOGIC1.3',
+                    'active' => true,
+                    'inStock' => 0,
+                    'lastStock' => true,
+                    'prices' => [
+                        [
+                            'customerGroupKey' => 'EK',
+                            'price' => 100,
+                        ],
+                    ],
+                    'configuratorOptions' => []
+                ]
+            ],
+            'filterGroupId' => 1,
+        ];
+
+        return [
+            'Out of stock variants are ignored' => [
+                'articleConfiguration' => $articleConfiguration,
+            ]
+        ];
+    }
+
+    /**
+     * @dataProvider variantPriceProvider*
+     *
+     * @param array $articleConfiguration
+     *
+     * @throws ReflectionException
+     */
+    public function testMainPriceNotConsideredWhenLastStock(array $articleConfiguration)
+    {
+        $articleFromConfiguration = $this->createTestProduct($articleConfiguration);
+
+        // Set lastStock value for Shopware <= 5.3
+        if (!method_exists($articleFromConfiguration->getMainDetail(), 'getLastStock')) {
+            /** @var Detail $detail*/
+            foreach ($articleFromConfiguration->getDetails() as $index => $detail) {
+                $article = (new Article())->setLastStock($articleConfiguration['variants'][$index]['lastStock']);
```

Comment:
You do not need `()` around your `Article` instance.

Reply:
I get the error `Parse error: syntax error, unexpected '->'` for `new Article()->setLastStock`. Or do you mean `(new Article)->setLastStock`?

Reply:
I see, that was actually my bad. I thought these brackets were going around the whole method.